payment/models.py

```diff
@@ -58,6 +60,53 @@ class PaymentRequest(ModelWithStr):
     callback = StringType()  # a webhook to call when a payment is complete
 
 
+class WhitelistRequest(ModelWithStr):
+    order_id = StringType()
+    source = StringType()
+    destination = StringType()
+    amount = IntType()
+    xdr = StringType()
+    network_id = StringType()
+    app_id = StringType()
+
+    @staticmethod
+    def compare_attr(attr1, attr2, attr_name):
+        if attr1 != attr2:
+            raise TransactionMismatch('{attr_name}: {attr1} does not match expected {attr_name}: {attr2}'.
+                                      format(attr_name=attr_name,
+                                             attr1=attr1,
+                                             attr2=attr2))
+
+    def verify_transaction(self):
+        """Verify that the encoded transaction matches our expectations"""
+        try:
+            decoded_tx = decode_transaction(self.xdr, self.network_id)
+        except Exception as e:
+            if isinstance(e, KinErrors.CantSimplifyError):
+                raise TransactionMismatch('Unexpected transaction')
+            log.error('Couldn\'t decode tx with xdr: {}'.format(self.xdr))
+            raise TransactionMismatch('Transaction could not be deocded')
+        memo_parts = decoded_tx.memo.split('-')
+        if len(memo_parts) != 3:
+            raise TransactionMismatch('Unexpected memo')
+        self.compare_attr(memo_parts[1], self.app_id, 'App id')
+        self.compare_attr(memo_parts[2], self.order_id, 'Order id')
+        self.compare_attr(decoded_tx.source, self.source, 'Source account')
+        self.compare_attr(decoded_tx.operation.destination, self.destination, 'Destination account')
+        self.compare_attr(decoded_tx.operation.amount, self.amount, 'Amount')
+
+    def whitelist(self) -> str:
+        """Sign and return a transaction to whitelist it"""
+        # Get app hot wallet account
+        # Fix for circular imports
+        # https://stackoverflow.com/questions/1250103/attributeerror-module-object-has-no-attribute
+        from .blockchain import get_sdk
+        app_seed = APP_SEEDS.get(self.app_id).our
+        with get_sdk(app_seed) as hot_account:
+            return hot_account.write_sdk.whitelist_transaction({'envelope': self.xdr,
```

Comment:
this API looks a bit like javascript, why are you sending a dictionary object and not plain args?

Reply:
"whitelist_transaction" expects a json string (but also knows how to handle a dict), since the client always sends the info in json and the server just passes the body to the function